bids/analysis/base.py

```diff
@@ -266,11 +266,11 @@ def get_Xy(self, drop_entities=True, **selectors):
         for name, g in data.groupby(ent_cols):
             ent_data = g[ent_cols].drop_duplicates().iloc[0, :]
             ents = ent_data.to_dict()
-            if 'run' in ent_cols:
-                img = self.analysis.layout.get(return_type='file', type='bold',
+            if 'onset' in data.columns:
+                img = self.analysis.layout.get(return_type='file', type='preproc',
```

Comment:
Not sure if this is generally desired, but you almost always want some kind of preprocessed data. We should also somehow permit users to specify other entities, such as `space='MNI...'`. (e.g. I might want to run a basic model in MNI, but also create a T1w space series for MVPA.)